### PR TITLE
Fixed validation for RepoUrlPicker

### LINKED ISSUE #25347: 🐛 Bug Report: Templates: Using the RepoUrlPicker leads to a ".repoUrl Incomplete repository location provided, owner not provided" validation error for Azure DevOps repositories since v1.28.0
### 📜 Description

When using a Template that uses the RepoUrlPicker to scaffold code in an Azure DevOps repository an validation error is thrown when validation the RepoUrlPicker field.

The validation error ".repoUrl Incomplete repository location provided, owner not provided" is shown since version v1.28.0

### 👍 Expected behavior

The validation shouldn't require the _owner_ for repoUrls constructed by the AzureRepoPicker

### 👎 Actual Behavior with Screenshots

A ".repoUrl Incomplete repository location provided, owner not provided" validation error is shown

### 👟 Reproduction steps

Create a scaffolding template where the RepoUrlPicker is used to select an Azure DevOps repository, see e.g.

```
   - title: Choose the location for your code
      required:
        - repoUrl
      properties:
        repoUrl:
          title: Repository Location
          type: string
          ui:field: RepoUrlPicker
          ui:options:
            allowedHosts:
              - dev.azure.com
```

Use the RepoUrlPicker to select an organization, project, and repo and validate the field by moving to the next step.

### 📃 Provide the context for the Bug.

This is due to #24798 where the AzureRepoPicker is changed to construct urls using the _project_ instead of _owner_ searchParam.

However the validation on the field requires the _owner_ searchParam to be present.

https://github.com/backstage/backstage/blob/a4cfe9d6f4d063bdb8d4e96a3470cf6dcae3e82b/plugins/scaffolder/src/components/fields/RepoUrlPicker/validation.ts#L59

Note that these changes made to the AzureRepoPicker where an url with _project_ instead of _owner_ is constructed is a breaking change, as these urls are used in, at least, the _scaffolder-node_ and _scaffolder-backend-module-azure_ plugins.

### 🖥️ Your Environment

_No response_

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

No, I don't have time to work on this right now

### PR DESCRIPTION
**Pull request by VeithBuergerhoff**
## Hey, I just made a Pull Request!

Fixes #25347

As stated in the issue, #24798 forgot to adjust the validation for the `owner`->`project` change.
This PR fixes that.

This is my first contribution to backstage. I do hope I used the changeset correctly.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/scaffolder-backend-module-azure/src/actions/azure.test.ts [hunk 1]
```diff
@@ -55,7 +55,9 @@ describe('publish:azure', () => {
   const action = createPublishAzureAction({ integrations, config });
 
   const mockContext = createMockActionContext({
-    input: { repoUrl: 'dev.azure.com?repo=repo&owner=owner&organization=org' },
+    input: {
```
Comment (VeithBuergerhoff) [inline plugins/scaffolder-backend-module-azure/src/actions/azure.test.ts:58]: Format changes were made by prettier

File: plugins/scaffolder/src/components/fields/RepoUrlPicker/validation.ts [hunk 2]
```diff
@@ -54,7 +54,14 @@ export const repoPickerValidation = (
           );
         }
       }
-      // For anything other than bitbucket and gerrit
+      if (integrationApi?.byHost(host)?.type === 'azure') {
```
Comment (VeithBuergerhoff) [inline plugins/scaffolder/src/components/fields/RepoUrlPicker/validation.ts:57]: This should be an else-if

I could have added the condition to the bitbucket check, but I prefer this

File: plugins/scaffolder-node/src/actions/util.ts [hunk 3]
```diff
@@ -85,6 +85,10 @@ export const parseRepoUrl = (
       checkRequiredParams(parsed, 'project', 'repo');
       break;
     }
+    case 'azure': {
```
Comment (VeithBuergerhoff) [inline plugins/scaffolder-node/src/actions/util.ts:88]: Could have been a fall-through, but a case for azure seems to be the pattern here

File: plugins/scaffolder/src/components/fields/RepoUrlPicker/validation.ts [hunk 4]
```diff
@@ -48,13 +48,19 @@ export const repoPickerValidation = (
           );
         }
 
+        if (!searchParams.get('project')) {
+          validation.addError(
+            'Incomplete repository location provided, project not provided',
+          );
+        }
+      } else if (integrationApi?.byHost(host)?.type === 'azure') {
```
Comment (VeithBuergerhoff) [inline plugins/scaffolder/src/components/fields/RepoUrlPicker/validation.ts:56]: I could have added the condition to the bitbucket check, but I prefer this

Comment (benjdlambert) [reply plugins/scaffolder/src/components/fields/RepoUrlPicker/validation.ts:56]: Can you add some tests in this `validation.test.ts` file for `azure` in this case :pray:

**Review by benjdlambert — CHANGES_REQUESTED** [review]
Hey :wave: thanks for this! Nice stuff!

Just a quick comment about adding some tests for this :pray:

**Comment by VeithBuergerhoff** [comment]
> Hey 👋 thanks for this! Nice stuff!
> 
> Just a quick comment about adding some tests for this 🙏

Done.
Please take a look at the tests I wrote and let me know if you want me to add/change anything.

File: plugins/scaffolder/src/components/fields/RepoUrlPicker/validation.ts (hunk 4 above)
Comment (VeithBuergerhoff) [reply plugins/scaffolder/src/components/fields/RepoUrlPicker/validation.ts:56]: Done.
Please take a look at the tests I wrote and let me know if you want me to add/change anything.

**Review by benjdlambert — APPROVED** [review]
Nice! thanks for the quick fix! :tada:

**Comment by Phiph** [comment]
Awesome thanks for the quick fix this <3 I stumbled on this in a demo I was doing today!

**Comment by ckuijjer** [comment]
Amazing! Thanks a lot for the quick fix!

**Comment by benjdlambert** [comment]
Thanks for this! Will go out in a `-next` release today, will look at pushing this in a patch release also I think.

**Pull request merged**